src/orion/plotting/_core.py

```diff
@@ -0,0 +1,85 @@
+import orion.plotting.backend_plotly as backend
+
+
+def regret(experiment, order_by='suggested', verbose_hover=False, **kwargs):
+    """
+    Make a plot to visualize the performance of the hyper-optimization process.
+
+    The x-axis contain the trials and the y-axis their respective best performance.
+
+    Parameters
+    ----------
+        experiment: ExperimentClient, Experiment or ExperimentView
+            The orion object containing the experiment data
+
+        order_by: str
+            Indicates how the trials should be ordered. Acceptable options are below.
+            See attributes of `Trial` for more details.
+
+            * 'suggested': Sort by trial suggested time (default).
+            * 'reserved': Sort by trial reserved time.
+            * 'completed': Sort by trial completed time.
+
+        verbose_hover: bool
+            Indicates whether to display the hyperparameter in hover tooltips. False by default.
+
+        **kwargs
+            All other plotting keyword arguments to be passed to
+            :meth:`plotly.express.line`.
+
+    Returns
+    -------
+    plotly.graph_objects.Figure
+
+    Raises
+    ------
+    ValueError
+        If no experiment is provided.
+
+    """
+    return backend.regret(experiment, order_by, verbose_hover, **kwargs)
+
+
+PLOT_METHODS = {'regret': regret}
```

Comment:
I think it would be worth reusing the same factory architecture that is used for Algorithms, Databases and Strategies. A backend could have a global configuration, hence it could be useful to represent it as an object instead of a module.

Reply:
I'm not sure what changes it would entail for the current source code related to plotting.

Reply:
Decision: The architecture will be changed when the Concept class will be integrated (~0.2.0)